src/tiled/changemapobject.cpp

```diff
@@ -158,3 +158,35 @@ void ChangeMapObjectsTile::changeTiles()
 
     emit mMapDocument->mapObjectModel()->objectsChanged(mMapObjects);
 }
+
+DetachObject::DetachObject(MapDocument *mapDocument,
+                           MapObject *mapObject)
+    : QUndoCommand(QCoreApplication::translate("Undo Commands",
+                                               "Detach Object",
+                                                nullptr))
+    , mMapDocument(mapDocument)
+    , mMapObject(mapObject)
+{
+    mTemplateRef = mMapObject->templateRef();
+    mProperties = mMapObject->properties();
+}
+
+void DetachObject::redo()
+{
+    // Merge the instance properties into the template properties
+    Properties newProperties = mMapObject->templateObject()->properties();
+    newProperties.merge(mMapObject->properties());
+    mMapObject->setProperties(newProperties);
+
+    mMapObject->setTemplateRef({nullptr, 0});
+    mMapObject->syncWithTemplate();
```

Comment:
Why do a `syncWithTemplate` call after setting the template reference to 0? Did you mean to do that before, just in case? I guess it can just be removed.

Reply:
I'm not sure why I added it but it does nothing.